portal/v2/src/ClusterDetailListComponents/ClusterOperatorsWrapper.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect, useRef } from "react"
+import { AxiosResponse } from 'axios';
+import { FetchClusterOperators } from '../Request';
+import { ICluster } from "../App"
+import { IMessageBarStyles, MessageBar, MessageBarType, Stack } from '@fluentui/react';
+import { clusterOperatorsKey } from "../ClusterDetail";
+import { ClusterOperatorListComponent } from "./ClusterOperatorList";
+
+export interface ICondition {
+  status: string,
+  reason: string,
+  lastUpdated: string,
+  message: string
+}
+
+export interface IClusterOperator {
+  name: string,
+  available: string,
+  conditions?: ICondition[],
+}
+
+export function ClusterOperatorsWrapper(props: {
+  currentCluster: ICluster
+  detailPanelSelected: string
+  loaded: boolean
+}) {
+  const [data, setData] = useState<any>([])
+  const [error, setError] = useState<AxiosResponse | null>(null)
+  const state = useRef<ClusterOperatorListComponent>(null)
+  
+  const [fetching, setFetching] = useState("")
+
+  const errorBarStyles: Partial<IMessageBarStyles> = { root: { marginBottom: 15 } }
+
+  const errorBar = (): any => {
+    return (
+      <MessageBar
+        messageBarType={MessageBarType.error}
+        isMultiline={false}
+        onDismiss={() => setError(null)}
+        dismissButtonAriaLabel="Close"
+        styles={errorBarStyles}
+      >
+        {error?.statusText}
+      </MessageBar>
+    )
+  }
+
+  // updateData - updates the state of the component
+  // can be used if we want a refresh button.
+  // api/clusterdetail returns a single item.
+  const updateData = (newData: any) => {
+    setData(newData)
+    const clusterOperatorList: IClusterOperator[] = []
+    if (state && state.current) {
+      newData.operators.forEach((element: { name: any;
+                                        available: any;
+                                        conditions: ICondition[]}) => {
+        const clusterOp: IClusterOperator = {
+          name: element.name,
+          available: element.available
+        }
+        clusterOp.conditions = []
+        element.conditions.forEach((condition: ICondition) => {
+          clusterOp.conditions!.push(condition)
+        });
+        clusterOperatorList.push(clusterOp)
+      });
+      state.current.setState({ clusterOperators: clusterOperatorList })
```

Comment:
Why are we using a `useRef` for this?

Reply:
I believe so that it can be passed down to the child component and updated

Reply:
I might add this to things to talk about on a react call.